### PR TITLE
ci: update ubuntu_workfow to use GITHUB_STEP_SUMMARY instead of posting comments

### PR DESCRIPTION
**Pull request by Lyokone**
Also updates gradle action according to this: https://github.com/gradle/actions/blob/main/docs/deprecation-upgrade-guide.md#the-action-gradlegradle-build-action-has-been-replaced-by-gradleactionssetup-gradle

Fixes #https://github.com/Flank/flank/issues/2544

### CODE REVIEW

**Review by exaby73 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -98,22 +99,16 @@ jobs:
         gh release upload $RELEASE_TAG flank.sha256
 
     - name: Gradle Upload to MavenCentral
-      uses: gradle/gradle-build-action@v3.5.0
       env:
         PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
-      with:
-        arguments: :test_runner:publishMavenJavaPublicationToMavenCentralRepository -PMVN_CENTRAL_USER=${{ secrets.MVN_CENTRAL_USER }} -PMVN_CENTRAL_PASSWORD=${{ secrets.MVN_CENTRAL_PASSWORD }}
+      run: ./gradlew :test_runner:publishMavenJavaPublicationToMavenCentralRepository -PMVN_CENTRAL_USER=${{ secrets.MVN_CENTRAL_USER }} -PMVN_CENTRAL_PASSWORD=${{ secrets.MVN_CENTRAL_PASSWORD }}
```
Comment (jsoref) [inline .github/workflows/release.yml:104]: You shouldn't use `${{ ...}}` in `run:`. 

Instead, best practice is: 

```yaml
env:
  MVN_CENTRAL_USER: ${{ secrets.MVN_CENTRAL_USER }}
  ... 
run: ... -PMVN_CENTRAL_USER="$MVN_CENTRAL_USER"... 
```

Where each secret/variable is filled into `env` and then used via an env variable.

File: .github/workflows/release_flank_scripts.yml [hunk 2]
```diff
@@ -25,9 +25,10 @@ jobs:
...
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v4
+
     - name: Gradle Upload to Github packages and Github release
-      uses: gradle/gradle-build-action@v3.5.0
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        arguments: "flank-scripts:releaseFlankScripts -PGITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}"
+      run: ./gradlew flank-scripts:releaseFlankScripts -PGITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
```
Comment (jsoref) [inline .github/workflows/release_flank_scripts.yml:34]: You already have `$GITHUB_TOKEN` available from `env`. 

That said, best practice is to rename that specific `env` variable `GH_TOKEN`. GitHub doesn't want people defining variables in their `GITHUB_... ` "namespace".

File: .github/workflows/ubuntu_workflow.yml [hunk 3]
```diff
@@ -41,17 +41,18 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           HEAD_REF: ${{ github.head_ref }}
-        with:
-          arguments: "clean build"
-
+        run: ./gradlew clean build
+      
   publish-scan-url:
-    needs: [ build ]
+    needs: [build]
```
Comment (jsoref) [inline .github/workflows/ubuntu_workflow.yml:55]: Personally, I prefer: 
```suggestion
    needs:
     - build
```
It makes it easier to add additional items and preserve blame.

File: .github/workflows/ubuntu_workflow.yml [hunk 4]
```diff
@@ -67,34 +68,13 @@ jobs:
...
-            **Timestamp:** ${{ steps.current-time.outputs.formattedTime }}
-            **Buildscan url for ubuntu-workflow run [${{ github.run_id }}](https://github.com/Flank/flank/actions/runs/${{ github.run_id }})**
-            ${{ needs.build.outputs.build-scan-url }}
+      - name: Write Buildscan URL to Summary
+        run: |
+          echo "**Timestamp:** ${{ steps.current-time.outputs.formattedTime }}" >> $GITHUB_STEP_SUMMARY
+          echo "**Buildscan URL for ubuntu-workflow run:**" >> $GITHUB_STEP_SUMMARY
+          echo "[${{ github.run_id }}](https://github.com/Flank/flank/actions/runs/${{ github.run_id }})" >> $GITHUB_STEP_SUMMARY
+          echo "${{ needs.build.outputs.build-scan-url }}" >> $GITHUB_STEP_SUMMARY
+
```
Comment (jsoref) [inline .github/workflows/ubuntu_workflow.yml:80]: You probably have trailing whitespace on this line. 


```suggestion

```

**Comment by Lyokone** [comment]
Thanks for the detailed review @jsoref :)

**Review by kaibolay — APPROVED** [review]
Thank you!

File: .github/workflows/ubuntu_workflow.yml [hunk 5]
```diff
@@ -77,5 +77,4 @@ jobs:
           echo "**Timestamp:** ${{ steps.current-time.outputs.formattedTime }}" >> $GITHUB_STEP_SUMMARY
           echo "**Buildscan URL for ubuntu-workflow run:**" >> $GITHUB_STEP_SUMMARY
           echo "[${{ github.run_id }}](https://github.com/Flank/flank/actions/runs/${{ github.run_id }})" >> $GITHUB_STEP_SUMMARY
-          echo "${{ needs.build.outputs.build-scan-url }}" >> $GITHUB_STEP_SUMMARY
-      
\ No newline at end of file
+          echo "${{ needs.build.outputs.build-scan-url }}" >> $GITHUB_STEP_SUMMARY
```
Comment (jsoref) [inline .github/workflows/ubuntu_workflow.yml:80]: This should have been changed to use an `env:`, and you're missing the blank line at the end of the file.

Comment (kaibolay) [reply .github/workflows/ubuntu_workflow.yml:80]: Oops. Can you send a PR? Thanks!

Comment (jsoref) [reply .github/workflows/ubuntu_workflow.yml:80]: Sure

Comment (jsoref) [reply .github/workflows/ubuntu_workflow.yml:80]: https://github.com/Flank/flank/pull/2547

**Pull request merged**